packages/slate/src/commands/at-range.js

```diff
@@ -375,23 +375,38 @@ Commands.deleteBackwardAtRange = (editor, range, n = 1) => {
 
   // Otherwise, we need to see how many nodes backwards to go.
   let node = text
-  let offset = 0
   let traversed = focus.offset
+  const prevSibling = document.getPreviousSibling(node.key)
 
-  while (n > traversed) {
-    node = document.getPreviousText(node.key)
-    const next = traversed + node.text.length
+  // If the next sibling is an inline, has no text and the focus offset is 0,
+  // then delete the inline
+  if (
+    traversed === 0 &&
```

Comment:
would you want to refactor this compound conditional to its own method? something like 'shouldDeleteInline'